modules/aws/ec2-files.go

```diff
@@ -218,22 +218,16 @@ func FetchFilesFromAsgsE(t *testing.T, awsRegion string, spec RemoteFileSpecific
 
 			instanceIDs, err := GetInstanceIdsForAsgE(t, curAsg, awsRegion)
 			if err != nil {
-				return err
-			}
-
-			errorsOccurred := []error{}
-
-			for _, instanceID := range instanceIDs {
-				err = FetchFilesFromInstanceE(t, awsRegion, spec.SshUser, spec.KeyPair, instanceID, spec.UseSudo, curRemoteDir, spec.LocalDestinationDir, fileFilters)
+				errorsOccurred = append(errorsOccurred, err)
+			} else {
+				for _, instanceID := range instanceIDs {
+					err = FetchFilesFromInstanceE(t, awsRegion, spec.SshUser, spec.KeyPair, instanceID, spec.UseSudo, curRemoteDir, spec.LocalDestinationDir, fileFilters)
```

Comment:
This is still reusing the `err` from before, but I suppose now it will happen to work... I still very much prefer immutable variables that never change from declaration to avoid these pitfalls, but this may be good enough for now.

Reply:
agreed